lib/Targets/AIETargetXAIEV2.cpp

```diff
@@ -844,6 +847,79 @@ mlir::LogicalResult AIETranslateToXAIEV2(ModuleOp module, raw_ostream &output) {
   for (auto lock : module.getOps<LockOp>())
     lockAccessor(lock);
 
+  //---------------------------------------------------------------------------
+  // Output ObjectFifo Accessors (Shim)
+  //---------------------------------------------------------------------------
+  auto objFifoAccessor = [&](LockOp lock) {
+    // for each lock generated from the objectFifo lowering,
+    // generate acquire / release accessors for producer and consumer sides
+    int col = lock.colIndex();
+    int row = lock.rowIndex();
+    if (!lock.hasName())
+      return;
+
+    if (row != 0)
+      return;
+
+    // naming scheme is: <objFifo_name>_<'prod' or 'cons'>_<lock_number>
+    std::string lockName(lock.name().getValue());
+    std::string prodLock = "_prod_lock_";
+    std::string consLock = "_cons_lock_";
+
+    std::string objFifoName;
+    std::string accessor_type;
+    std::string objIndex;
+    int acquireLockValue = -1;
+    int releaseLockValue = -1;
+
+    auto pos = lockName.find(prodLock);
+    if (pos != std::string::npos) {
+      // Producer
+      objFifoName = lockName.substr(0, pos);
+      objIndex = lockName.substr(pos + prodLock.length());
+      accessor_type = "producer";
+      acquireLockValue = 0;
+      releaseLockValue = 1;
```

Comment:
I'm not a fan of parsing the name here.   Why isn't the full lock name put the sym_name of the lock, so that the regular handling of locks can happen at this level?

Reply:
I tried to do that initially but I hit a roadblock when passing the aie context from the function definition into the lock accessor function call.

Reply:
I'm also -1 on parsing the symbol name. Consider the case where a non-objfifo user has your string in their symbol name. Can you use some metadata (attribute) other than the symbol name?